### PR TITLE
Add failing test case for urgent strategy and default scope

### PR DESCRIPTION
**Pull request by gkop**
This is an "interesting" regression I noticed since the work refactoring the strategies.

We have different behavior between urgent and atomic strategy: atomic strategy passes id backrefs, while urgent strategy passes the objects themselves.

This ends up causing the atomic strategy updates to respect the default activerecord scope, while the urgent strategy updates do not: https://github.com/toptal/chewy/blob/0c726fb79912817e662ac432ead978ad33fce88a/lib/chewy/type/adapter/active_record.rb#L74-L80 .

The easiest fix might be only using id backrefs in this case in ActiveRecord#import, but the existing test suite supports indexing records for which id is nil, so I cannot make this fix on my own since it will invalidate many existing tests.

Also it occurred to me to repeat the default scope logic when passing the backref in the model observing block, but this doesn't work: in this block we can only decide whether or not to update the index for the backref, not whether to reindex or delete. So if a record goes from in scope to out of scope, we must be observing it in order to fulfill the necessary deletion.

So finally I am left with the fix I propose in this PR, modifying the urgent strategy. It's not a very satisfying fix since it doesn't address the root of the problem. But perhaps it's better than the current inconsistency between strategies.

### CODE REVIEW

**Comment by gkop** [comment]
Re: mongoid, I ran `respec -t mongoid` to try to replicate the issue in travis but got a lot of strange errors:

```
     Failure/Error: specify { expect(PlacesIndex.limit(6).load(city: { scope: ->{ where(:rating.lt => 2) } }).order(:rating))
     NoMethodError:
       undefined method `to_sym' for #<Origin::Key:0x0000000680ff88>
```

If you want to eventually merge this test case in, I can surely make it work with mongoid's named scopes.

**Comment by pyromaniac** [comment]
Hey, thanks for the patch, but I have to think about it. Actually, it was never supposed to use conditions in default scope. Instead there is `delete_from_index` method which might be defined in model to specify such conditions.

After your PR I've got 2 thoughts:
1. It would be more convenient to define delete_from_index or similar method not in model, but in chewy type since there might be multiple types defined for one model.
2. Chewy should extract conditions from default scope to use them exactly the same way as delete_from_index method, but this would work in simple cases only.

So thanks for pointing on the problem, I'll think about it a little bit.

File: lib/chewy/strategy/urgent.rb [hunk 1]
```diff
@@ -10,7 +10,12 @@ class Strategy
     #
     class Urgent < Base
       def update type, objects, options = {}
-        type.import(objects)
+        objects = Array(objects)
```
Comment (pyromaniac) [inline lib/chewy/strategy/urgent.rb]: Could I ask you to replace this with Array.wrap ?

**Comment by pyromaniac** [comment]
I've decided to force import to respect default type scope, so this PR is not relevant anymore. Could you try master please?

**Pull request closed without merging**